javacord-api/src/main/java/org/javacord/api/entity/channel/TextableRegularServerChannelUpdater.java

```diff
@@ -0,0 +1,85 @@
+package org.javacord.api.entity.channel;
+
+import org.javacord.api.entity.channel.internal.TextableRegularServerChannelUpdaterDelegate;
+import org.javacord.api.util.internal.DelegateFactory;
+
+public class TextableRegularServerChannelUpdater<T extends TextableRegularServerChannelUpdater<T>>
+        extends RegularServerChannelUpdater<T> {
+
+    /**
+     * The channel delegate used by this instance.
+     */
+    protected final TextableRegularServerChannelUpdaterDelegate textableRegularServerChannelUpdaterDelegate;
+
+    /**
+     * Creates a new textable regular server channel updater.
+     *
+     * @param delegate A subtype of a RegularServerChannelUpdaterDelegate.
+     */
+    protected TextableRegularServerChannelUpdater(TextableRegularServerChannelUpdaterDelegate delegate) {
+        super(delegate);
+        this.textableRegularServerChannelUpdaterDelegate = delegate;
+    }
+
+    /**
+     * Creates a new textable regular server channel updater.
+     *
+     * @param channel The channel to update.
+     */
+    public TextableRegularServerChannelUpdater(TextableRegularServerChannel channel) {
+        this(DelegateFactory.createTextableRegularServerChannelUpdaterDelegate(channel));
+    }
+
+    /**
+     * Queues the category to be updated.
+     *
+     * @param category The new category of the channel.
+     * @return The current instance in order to chain call methods.
+     */
+    public TextableRegularServerChannelUpdater<T> setCategory(ChannelCategory category) {
```

Comment:
The same for the other methods
```suggestion
    public T setCategory(ChannelCategory category) {
```